src/LuaTransformer.ts

```diff
@@ -3853,12 +3854,32 @@ export class LuaTransformer {
             } else if (ts.isMethodDeclaration(element)) {
                 const expression = this.transformFunctionExpression(element);
                 properties.push(tstl.createTableFieldExpression(expression, name, element));
+            } else if (ts.isSpreadAssignment(element)) {
+                if (properties.length > 0) {
+                    const tableExpression = tstl.createTableExpression(properties, expression);
+                    tableExpressions.push(tableExpression);
+                }
+                properties = [];
```

Comment:
You already handle this situation (line 3858-3862) in lines 3872-3875. I think you can remove these 5 lines.

Reply:
Alright so the current algorithm creates multiple tables in the order the properties are declared so if they are overwritten they are overwritten in the correct order.

Source

```ts
let obj = { x: 0, y: 1, z: 2 };
const object = { x: 0, ...obj, y: 1, z: 2 }
const o = { x: true, ...{ y: true, z: true } };
```

Output with this code:

```diff
  local obj = {x = 0, y = 1, z = 2}
+ local object = __TS__ObjectAssign({x = 0}, obj, {y = 1, z = 2})
+ local o = __TS__ObjectAssign({x = true}, {y = true, z = true})
```

Output without:

```diff
  local obj = {x = 0, y = 1, z = 2}
- local object = __TS__ObjectAssign({}, obj, {x = 0, y = 1, z = 2})
- local o = __TS__ObjectAssign({y = true, z = true}, {x = true})
```

Reply:
I see, that makes sense. Could you place a comment on the first occurrence explaining that? Something like `// Create table for preceding properties to preserve order of elements.`?